app/SmartAlbums/HasSimpleRelationships.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+namespace App\SmartAlbums;
+
+/**
+ * Trait HasSimpleRelationships.
+ *
+ * This trait is a copy of
+ * {@link \Illuminate\Database\Eloquent\Concerns\HasRelationships}
+ * stripped down to the essential methods which are required by a class to
+ * own relationships without being a full model.
```

Comment:
A totally agree with you.

Actually, I do not like the whole architecture. It really bothers me that I had to copy & paste the code. Previously, the problem was hidden under the covers because all smart albums inherited from `Album` which of course was not good either.

The underlying problem actually is actually twofold:

 1. The completely weird way how Laraval uses PHP traits. Normally, a trait should be "closed", i.e. it should not know by which class it is used and should not depend on other traits being used by the same class.
 2. That Eloquent's class `Relation` requires an object of class `Model` although it does not use a single method which is specific to `Model`.

It is barely impossible to implement a clean architecture based on a framework whose on architecture is fundamentally broken. OK, enough rant about Laravel.

The initial reasons why I tried to make the smart albums mimic the behavior of `Model` were

 1. higher consistency to the old design
 2. identical treatment of thumbnails; the thumbnail logic assumed that every `AbstractAlbum` has a method `photos()` which returns a `Relation`-object

The first is a weak argument. The second argument has become outdated, because I recently have implemented a special thumbnail relationship for albums, so that normal albums and smart albums are different wrt. this aspect anyway.

Give me some time to think about it, but probably I will just dump the whole code.